### PR TITLE
feat: POC for ddev config --update

### PR DESCRIPTION
**Pull request by penyaskito**
## How This PR Solves The Issue

For supporting discussion on https://github.com/ddev/ddev/issues/6035.

If we end up going this path I'd create a cleaner PR.

### CODE REVIEW

**Comment by rfay** [comment]
WOW!

File: cmd/ddev/cmd/config.go [hunk 1]
```diff
@@ -368,6 +373,63 @@ func handleMainConfigArgs(cmd *cobra.Command, _ []string, app *ddevapp.DdevApp)
...
+		output.UserOut.Print(detectedApptypeMsg)
+
+		// For the POC we need ConfigFileOverrideAction to omit app.ConfigExists() check
+		err = app.ConfigFileOverrideAction()
+
+		apptypeMsg := fmt.Sprintf("Apptype is %s", app.Type)
+		output.UserOut.Print(apptypeMsg)
+
+		if app.Type == nodeps.AppTypeDrupal {
+			// I can't call getDrupalVersion from drupal.go here?
```
Comment (penyaskito) [inline cmd/ddev/cmd/config.go:391]: TIL lowercase/uppercase defines visibility. So needs to be renamed to GetDrupalVersion if we want this function to be exported. https://go.dev/tour/basics/3

Comment (rfay) [reply cmd/ddev/cmd/config.go:391]: Your IDE can easily do this with a refactor action.

File: cmd/ddev/cmd/config.go [hunk 2]
```diff
@@ -368,6 +373,63 @@ func handleMainConfigArgs(cmd *cobra.Command, _ []string, app *ddevapp.DdevApp)
...
+		detectedApptypeMsg := fmt.Sprintf("Detected %s", detectedApptype)
+		output.UserOut.Print(detectedApptypeMsg)
+
+		// For the POC we need ConfigFileOverrideAction to omit app.ConfigExists() check
+		err = app.ConfigFileOverrideAction()
+
+		apptypeMsg := fmt.Sprintf("Apptype is %s", app.Type)
+		output.UserOut.Print(apptypeMsg)
+
+		if app.Type == nodeps.AppTypeDrupal {
```
Comment (rfay) [inline cmd/ddev/cmd/config.go:390]: I don't think you should need to mention specific project types here.

**Review by rfay — CHANGES_REQUESTED** [review]
The way I was thinking about this was that it would just be like any other config action (including --auto), except that it would not respect existing config. The key there is one line IIRC. Will take a look.

File: cmd/ddev/cmd/config.go (hunk 2 above)
Comment (penyaskito) [reply cmd/ddev/cmd/config.go:405]: Yeah, it was only for debugging purposes when I did that. I just thought that we might want to take this chance to change apptype drupal8, drupal9, drupal10 to drupal if we find those.

**Comment by penyaskito** [comment]
> Essentially we need to do what it was doing before that... but only on update?

Oh right I was accidentally hit by that one! Makes sense, it would be probably cleaner.

One question I have if we go either path is: should we really change db? That might be dangerous. Even if I don't think data loss could happen, people might not find their data if the db versions change. I suggest if a change in db is found, we leave it as is and provide instructions on updating/migrating it.

File: pkg/ddevapp/apptypes.go [hunk 3]
```diff
@@ -388,9 +388,9 @@ func (app *DdevApp) PostImportDBAction() error {
 }
 
 // ConfigFileOverrideAction gives a chance for an apptype to override any element
-// of config.yaml that it needs to (on initial creation, but not after that)
-func (app *DdevApp) ConfigFileOverrideAction() error {
-	if appFuncs, ok := appTypeMatrix[app.Type]; ok && appFuncs.configOverrideAction != nil && !app.ConfigExists() {
+// of config.yaml that it needs to
+func (app *DdevApp) ConfigFileOverrideAction(overrideExistingConfig bool) error {
+	if appFuncs, ok := appTypeMatrix[app.Type]; ok && appFuncs.configOverrideAction != nil && (overrideExistingConfig || !app.ConfigExists()) {
```
Comment (penyaskito) [inline pkg/ddevapp/apptypes.go:393]: @rfay Even if we go the other path, we probably still need this.

**Comment by rfay** [comment]
> if we go either path is: should we really change db

We should not change database type or version. It would make people crazy. Thanks for paying attention to this. 

I see that this would affect CraftCMS, which really wants MySQL 8.0. But I don't see why we'd change existing.

I see drupal8/9 have override to mariadb:10.4, but I don't think those belong in there in the first place? Why would we not use defaults?

Silverstripe is probably a mistake. @GuySartorelli should we be overriding database type and version in https://github.com/ddev/ddev/blob/64c9e469427785debfdda7db2e8caf1b8632433c/pkg/ddevapp/silverstripe.go#L78-L79 ? For that matter, should we even have silverstripeConfigOverrideAction or should it be removed?

**Comment by penyaskito** [comment]
@rfay I've created https://github.com/ddev/ddev/pull/6055. It's definitively cleaner that way. I'll leave this one open still in case we want to keep these conversations here, but feel free to close it if you feel so.

**Comment by rfay** [comment]
Closing just to avoid confusion, can be reopened.

**Comment by GuySartorelli** [comment]
I recognise this was closed but just answering what was asked:

> Silverstripe is probably a mistake. @GuySartorelli should we be overriding database type and version in? For that matter, should we even have silverstripeConfigOverrideAction or should it be removed?

I'm a bit removed from the workflow of someone who uses Silverstripe CMS to make websites - my workflow is very specific to the maintenance work I do so I'm not sure whether this would be a useful or harmful.

It is worth noting that there have been a few discussions about how DDEV overrides values in `.env` (e.g. database name, user, and password) which some people have found to disrupt their preferred workflow. I think that's a separate concern, but worth mentioning. I actually explicitly avoid using the "silverstripe" project type because of it.

**Pull request closed without merging**